xds/internal/xdsclient/client.go

```diff
@@ -733,6 +742,35 @@ func (c *clientImpl) Close() {
 	c.logger.Infof("Shutdown")
 }
 
+func (c *clientImpl) updateValidator(u interface{}) error {
+	switch update := u.(type) {
+	case *SecurityConfig:
```

Comment:
This is checking for a subtype, not the top level updates.
The concern is, it will be hard to track where this function is called, and also easy to miss a place this should be called.

Should we limit this to only the top level updates?
Are the fields accessible from LDS and CDS update structs?

Reply:
The `SecurityConfig` is hidden deep within the `ListenerUpdate` on the server side. That was the reason I decided to take this approach. But after your comment about "easy to miss a place this should be called", I re-tried that, and doesn't look very bad. Thanks.